src/test/test-models/good/int_overloads.stan

```diff
@@ -0,0 +1,194 @@
+  transformed_data_real = expm1(d_int);
+  transformed_data_real = expm1(d_real);
+  transformed_data_real = log(d_int);
+  transformed_data_real = log(d_real);
+  transformed_data_real = log1p(d_int);
+  transformed_data_real = log1p(d_real);
+  transformed_data_real = log2(d_int);
+  transformed_data_real = log2(d_real);
+  transformed_data_real = log10(d_int);
+  transformed_data_real = log10(d_real);
+
+  transformed_data_real = pow(d_int);
+  transformed_data_real = pow(d_real);
+  transformed_data_real = sqrt(d_int);
+  transformed_data_real = sqrt(d_real);
+  transformed_data_real = cbrt(d_int);
+  transformed_data_real = cbrt(d_real);
+
+  transformed_data_real = sin(d_int);
+  transformed_data_real = sin(d_real);
+  transformed_data_real = cos(d_int);
+  transformed_data_real = cos(d_real);
+  transformed_data_real = tan(d_int);
+  transformed_data_real = tan(d_real);
+  transformed_data_real = asin(d_int);
+  transformed_data_real = asin(d_real);
+  transformed_data_real = acos(d_int);
+  transformed_data_real = acos(d_real);
+  transformed_data_real = atan(d_int);
+  transformed_data_real = atan(d_real);
+
+  transformed_data_real = sinh(d_int);
+  transformed_data_real = sinh(d_real);
+  transformed_data_real = cosh(d_int);
+  transformed_data_real = cosh(d_real);
+  transformed_data_real = tanh(d_int);
+  transformed_data_real = tanh(d_real);
+  transformed_data_real = asinh(d_int);
+  transformed_data_real = asinh(d_real);
+  transformed_data_real = acosh(d_int);
+  transformed_data_real = acosh(d_real);
+  transformed_data_real = atanh(d_int);
+  transformed_data_real = atanh(d_real);
+
+  transformed_data_real = erf(d_int);
+  transformed_data_real = erf(d_real);
+  transformed_data_real = erfc(d_int);
+  transformed_data_real = erfc(d_real);
+  transformed_data_real = tgamma(d_int);
+  transformed_data_real = tgamma(d_real);
+
+  transformed_data_real = ceil(d_int);
+  transformed_data_real = ceil(d_real);
+  transformed_data_real = floor(d_int);
+  transformed_data_real = floor(d_real);
+  transformed_data_real = trunc(d_int);
+  transformed_data_real = trunc(d_real);
+  transformed_data_real = round(d_int);
+  transformed_data_real = round(d_real);
+
+  transformed_data_real = inv(d_int);
+  transformed_data_real = inv(d_real);
+  transformed_data_real = inv_sqrt(d_int);
+  transformed_data_real = inv_sqrt(d_real);
+  transformed_data_real = inv_square(d_int);
+  transformed_data_real = inv_square(d_real);
+  transformed_data_real = sqrt(d_int);
+  transformed_data_real = sqrt(d_real);
+  transformed_data_real = tan(d_int);
+  transformed_data_real = tan(d_real);
+  transformed_data_real = tanh(d_int);
+  transformed_data_real = tanh(d_real);
+  transformed_data_real = trigamma(d_int);
+  transformed_data_real = trigamma(d_real);
+}
+
+parameters {
+  real p_real;
+  real y_p;
+}
+model {
+  y ~ normal(0,1);
+}
+
+transformed parameters {
+  real transformed_param_real;
+
+  transformed_param_real = abs(d_int);
```

Comment:
These function applications are all already tested in the data section.  These should all be removed and replaced with ones with autodiff variable arguments.  In the case of `pow` and `atan2`, ones that do it in all combinations with `double` and `real`).

Reply:
Added.